src/EntityFactory.ts

```diff
@@ -21,8 +18,8 @@ export type EntityFactoryType =
 /**
  * Stores definitions of entities to allow for easy creation/cloning
  */
-export class EntityFactory {
-	entities: Map<EntityReference, any>;
+export class EntityFactory<T extends any> {
```

Comment:
Added types here to have methods like 'getDefinition' return the proper type being stored in factory.

Reply:
Is `extends any` what we need/want here? Or do we want an undefined `T` to default to `any`?